### PR TITLE
[invalid-class-object] Fix crash when ``__class__`` is defined with a tuple

### LINKED ISSUE #7467: Even after #7429 merged, a crash AttributeError: 'ClassDef' object has no attribute 'value'
### Bug description

When parsing the following file, Pylint crashes:

```python
class AnotherClass:
    ...


class Pylint7429:
    def foo(self):
        self.__class__, myvar = AnotherClass, "myvalue"
```

Pylint versions prior to 2.9 run fine on the same file.

### Configuration

_No response_

### Command used

```shell
pylint /tmp/pylint7429.py
```

### Pylint output

```python
************* Module pylint7429
[...]
Exception on node <AssignAttr.__class__ l.7 at 0x7fb5f19060d0> in file '/tmp/pylint7429.py'
Traceback (most recent call last):
  File "pylint/utils/ast_walker.py", line 90, in walk
    callback(astroid)
  File "pylint/checkers/classes/class_checker.py", line 1556, in visit_assignattr
    self._check_invalid_class_object(node)
  File "pylint/checkers/classes/class_checker.py", line 1561, in _check_invalid_class_object
    inferred = safe_infer(node.parent.value)
  File "astroid/bases.py", line 134, in __getattr__
    return getattr(self._proxied, name)
AttributeError: 'ClassDef' object has no attribute 'value'
/tmp/pylint7429.py:1:0: F0002: /tmp/pylint7429.py: Fatal error while checking '/tmp/pylint7429.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/home/akaihola/.cache/pylint/pylint-crash-2022-09-15-14-06-01.txt'. (astroid-error)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

Expect pylint to not crash

### Pylint version

```shell
pylint 2.15.2
astroid 2.12.9
Python 3.8.13 (default, Mar 16 2022, 13:02:57) 
[GCC 11.3.0]
```

### OS / Environment

NixOS 22.05

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #7467

### CODE REVIEW

File: doc/whatsnew/fragments/7467.bugfix [hunk 1]
```diff
@@ -0,0 +1,3 @@
+``invalid-class-object`` do not crash anymore when ``__class__`` is assigned alongside another variable.
```
Comment (jacobtylerwalls) [inline doc/whatsnew/fragments/7467.bugfix:1]:
```suggestion
``invalid-class-object`` does not crash anymore when ``__class__`` is assigned alongside another variable.
```

File: pylint/checkers/classes/class_checker.py [hunk 2]
```diff
@@ -684,7 +684,7 @@ def _has_same_layout_slots(
         "Used when a value in __slots__ conflicts with a class variable, property or method.",
     ),
     "E0243": (
-        "Invalid __class__ object",
+        "Invalid assignement to '__class__'. Should be a class definition but got a '%s'",
```
Comment (jacobtylerwalls) [inline pylint/checkers/classes/class_checker.py:687]:
```suggestion
        "Invalid assignment to '__class__'. Should be a class definition but got a '%s'",
```

File: pylint/checkers/classes/class_checker.py [hunk 3]
```diff
@@ -1560,15 +1560,31 @@ def visit_assignattr(self, node: nodes.AssignAttr) -> None:
...
             return
-        inferred = safe_infer(node.parent.value)
+        if isinstance(node.parent, nodes.Tuple):
+            class_index = -1
+            for i, elt in enumerate(node.parent.elts):
+                if hasattr(elt, "attrname") and elt.attrname == "__class__":
+                    class_index = i
+            if class_index == -1:
+                # This should not happen because we checked that the node name
+                # is '__class__' earlier, but let's not be too confident here
```
Comment (jacobtylerwalls) [inline pylint/checkers/classes/class_checker.py:1570]: Good to have this. Maybe assignment via nested tuples could have violated this assumption?

**Review by jacobtylerwalls — DISMISSED** [review]

File: pylint/checkers/classes/class_checker.py (hunk 2 above)
Comment (Pierre-Sassoulas) [reply pylint/checkers/classes/class_checker.py:687]: Hurgh, thank  you !

File: tests/functional/i/invalid/invalid_class_object.txt [hunk 4]
```diff
@@ -1,2 +1,5 @@
-invalid-class-object:17:0:17:11::Invalid __class__ object:UNDEFINED
-invalid-class-object:18:0:18:11::Invalid __class__ object:UNDEFINED
+invalid-class-object:20:0:20:11::Invalid assignement to '__class__'. Should be a class definition but got a 'Instance':INFERENCE
```
Comment (jacobtylerwalls) [inline tests/functional/i/invalid/invalid_class_object.txt:1]: These have to be updated.

Comment (Pierre-Sassoulas) [reply tests/functional/i/invalid/invalid_class_object.txt]: 🤦

**Review by jacobtylerwalls — APPROVED** [review]

**Pull request merged**